File: app/src/main/java/com/wxm/camerajob/data/db/DBOrmLiteHelper.java
package com.wxm.camerajob.data.db;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.util.Log;

import com.j256.ormlite.android.apptools.OrmLiteSqliteOpenHelper;
import com.j256.ormlite.dao.RuntimeExceptionDao;
import com.j256.ormlite.support.ConnectionSource;
import com.j256.ormlite.table.TableUtils;
import com.wxm.camerajob.data.define.CameraJob;
import com.wxm.camerajob.data.define.CameraJobStatus;
import com.wxm.camerajob.utility.FileLogger;

import java.sql.SQLException;

import wxm.androidutil.util.UtilFun;

/**
 * APP的sqlite helper
 * Created by wxm on 2016/8/12.
 */
public class DBOrmLiteHelper  extends OrmLiteSqliteOpenHelper {
    private static final String     TAG                 = "DBOrmLiteHelper";
    private static final String     DATABASE_NAME       = "AppLocal.db";
    private static final int        DATABASE_VERSION    = 6;


    private RuntimeExceptionDao<CameraJob, Integer> mRDAOCameraJob = null;
    private RuntimeExceptionDao<CameraJobStatus, Integer> mRDAOCameraJobStatus = null;


    public DBOrmLiteHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db, ConnectionSource connectionSource) {
        CreateAndInitTable();
    }


    @Override
    public void onUpgrade(SQLiteDatabase database,
                          ConnectionSource connectionSource, int oldVersion, int newVersion) {
        switch (newVersion) {
            case 5 :    {
                CreateAndInitTable();
            }
            break;

            case 6 :    {
                try {
                    TableUtils.dropTable(connectionSource, CameraJob.class, false);
                    TableUtils.dropTable(connectionSource, CameraJobStatus.class, false);
                } catch (SQLException e) {
                    e.printStackTrace();
                    FileLogger.getLogger().severe(
                            "dropTable failure, ex = " + UtilFun.ExceptionToString(e));
                }
                CreateAndInitTable();
            }
            break;
        }
    }

    @Override
    public void close() {
        super.close();
        mRDAOCameraJob       = null;
        mRDAOCameraJobStatus = null;
    }

    RuntimeExceptionDao<CameraJob, Integer> getCamerJobREDao()   {
        if(!isOpen())
            return null;

        if(null == mRDAOCameraJob)
            mRDAOCameraJob = getRuntimeExceptionDao(CameraJob.class);

        return mRDAOCameraJob;
    }

    RuntimeExceptionDao<CameraJobStatus, Integer> getCamerJobStatusREDao()   {
        if(!isOpen())
            return null;

        if(null == mRDAOCameraJobStatus)
            mRDAOCameraJobStatus = getRuntimeExceptionDao(CameraJobStatus.class);

        return mRDAOCameraJobStatus;
    }



    private void CreateAndInitTable()   {
        try {
            TableUtils.createTable(connectionSource, CameraJob.class);
            TableUtils.createTable(connectionSource, CameraJobStatus.class);
        } catch (SQLException e) {
            Log.e(TAG, "Can't create database", e);
            throw new RuntimeException(e);
        }
    }
}

File: app/src/main/java/com/wxm/camerajob/data/db/DBOrmLiteHelper.kt
package com.wxm.camerajob.data.db

import android.content.Context
import android.database.sqlite.SQLiteDatabase

import com.j256.ormlite.android.apptools.OrmLiteSqliteOpenHelper
import com.j256.ormlite.dao.RuntimeExceptionDao
import com.j256.ormlite.support.ConnectionSource
import com.j256.ormlite.table.TableUtils
import com.wxm.camerajob.data.entity.CameraJob
import wxm.androidutil.log.TagLog

import java.sql.SQLException

/**
 * sqlite helper from app
 * Created by wxm on 2016/8/12.
 */
class DBOrmLiteHelper(context: Context) : OrmLiteSqliteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {
    val cameraJobREDao: RuntimeExceptionDao<CameraJob, Int> = getRuntimeExceptionDao(CameraJob::class.java)

    override fun onCreate(db: SQLiteDatabase, connectionSource: ConnectionSource) {
        createAndInitTable()
    }

    override fun onUpgrade(database: SQLiteDatabase,
                           connectionSource: ConnectionSource, oldVersion: Int, newVersion: Int) {
        when (newVersion) {
            8 -> {
                try {
                    TableUtils.dropTable<CameraJob, Any>(connectionSource, CameraJob::class.java, false)
                } catch (e: SQLException) {
                    TagLog.e("Can't create database", e)
                    throw RuntimeException(e)
                }

                createAndInitTable()
            }
        }
    }

    private fun createAndInitTable() {
        try {
            TableUtils.createTable(connectionSource, CameraJob::class.java)
        } catch (e: SQLException) {
            TagLog.e("Can't create database", e)
            throw RuntimeException(e)
        }
    }

    companion object {
        private const val DATABASE_NAME = "AppLocal.db"
        private const val DATABASE_VERSION = 8
    }
}
